src/openrct2-ui/windows/Tooltip.cpp

```diff
@@ -70,7 +70,11 @@ void window_tooltip_show(const OpenRCT2String& message, ScreenCoordsXY screenCoo
     if (w != nullptr)
         return;
 
-    int32_t textWidth = FormatTextForTooltip(message);
+    OpenRCT2String formattedMessage;
+    formattedMessage.str = STR_STRING_DEFINED_TOOLTIP;
+    formattedMessage.args = Formatter();
+    formattedMessage.args.Add<rct_string_id>(message.str);
```

Comment:
I don't know if any tooltips use arguments, but you are losing whatever was in message.args which might be an issue.

Reply:
I've added an if statement. I am not sure if used NumBytes() funtion was intended for such tasks.

Reply:
Now tooltips with arguments don't have the right format: BLACK SMALL. You need to copy the arguments across to the formatted one.

Reply:
If I do the following (basically if statement for message without arguments):
```
formattedMessage.str = STR_STRING_DEFINED_TOOLTIP;
formattedMessage.args = message.args;
formattedMessage.args.Add<rct_string_id>(message.str);
```
then the arguments are copied from message and string should be formatted as BLACK SMALL.
I would like to understand working of `formattedMessage.args.Add<rct_string_id>(message.str);` better. Specifically if this funcion (`.Add<>()`) concatenates given value of `.args` with anything given in braces of `.Add<>()` or simply overwrites `.args`?
I am asking because there is `std::memcpy` used in template of this function and as far as I understand the target buffer will be overwritten.

Reply:
I think it would be better to do something like:
```
format_string(buffer1, sizeof(buffer1), message.str, message.args.Data());
FormatString(buffer1, sizeof(buffer1), "{BLACK}{SMALL}{STRING}", buffer1);
```

Have a look at what happens in `FormatTextForTooltip`

Reply:
I've looked there, but needed your suggestion to have that click in my head :)